osvimdriver/openstack/neutron/driver.py

```diff
@@ -19,14 +22,29 @@ def get_network_by_id(self, network_id):
             raise ValueError('network_id must be provided')
         neutron_client = self.__get_neutron_client()
         logger.debug('Retrieving network with id %s', network_id)
-        result = neutron_client.show_network(network_id)
-        return result['network']
+        try:
+            external_request_id = str(uuid.uuid4())
+
+            self._generate_additional_logs(network_id, 'sent', external_request_id, 'application/json',
+                                       'request', 'http', {'method' : 'get', 'uri' : LOG_URI_PREFIX +'/stacks'}, None)
+            result = neutron_client.show_network(network_id)
+            self._generate_additional_logs(result, 'received', external_request_id, 'application/json',
+                                       'response', 'http', {'status_code' : 200, 'status_reason_phrase' : 'ok'}, None)  
+            return result['network']
+        except (neutronexceptions.BadRequest, neutronexceptions.NotFound)  as e:
+            self._generate_additional_logs(e, 'received', external_request_id, 'plain/text',
+                                       'response', 'http', {'status_code' : e.code,'status_reason_phrase' : e.message }, None)
+            raise e
+
 
     def get_network_by_name(self, network_name):
         if network_name is None:
             raise ValueError('network_name must be provided')
         neutron_client = self.__get_neutron_client()
         logger.debug('Retrieving network with name %s', network_name)
+        external_request_id = str(uuid.uuid4())
+        self._generate_additional_logs(network_name, 'sent', external_request_id, 'application/json',
```

Comment:
Please check how the library sends the payload, if it in json format, the message field (natwork_name) must be put kept in Json format.